techlibs/xilinx/synth_xilinx.cc

```diff
@@ -410,6 +434,17 @@ struct SynthXilinxPass : public ScriptPass
 			run("clean");
 		}
 
+		if (check_label("finalize")) {
+			bool do_iopads = iopads || (ise && !noiopads);
+			if (help_mode || do_iopads)
+				run("clkbufmap -buf BUFG O:I -inpad IBUFG O:I", "(-inpad passed if '-iopads' or '-ise' and not '-noiopads')");
```

Comment:
Is there any harm in specifying `-inpad IBUFG O:I` all the time? For example, what if the user instantiated some IBUFGs, but did not want any extra tool promotion to happen?

Reply:
I don't quite follow? If pad insertion is disbled, that's presumably because the user wants some kind of out-of-context mode, and we definitely shouldn't insert the IBUFG.  On the other hand, if pad insertion is enabled, and the user wants to avoid IBUFG, they could just manually instantiate another pad type and/or manually instantiate just the BUFG (which will trigger plain IBUF insertion in the following iopadmap pass).

Or do you mean you want an option to disable BUFG insertion altogether?  I agree that would be useful, I'm going to add it.

Reply:
Sorry, I misread the use of this flag! But allow clock promotion to be disabled is a good thing to do.